src/app/components/add-card/add-card.component.ts

```diff
@@ -81,6 +81,16 @@ export class AddCardComponent implements OnDestroy {
     this.newTask = '';
   }
 
+  onTaskButtonClick(event: {id: number, dataJob: string}): void {
+    const id = event.id;
+    const dataJob = event.dataJob;
+    const taskIndex = this.taskService.taskList.findIndex(task => task.id === id);
+
+    if (taskIndex !== -1 && dataJob === this.constantsService.DELETE_TODO) {
+      this.taskService.taskList.splice(taskIndex, 1);
+    }
```

Comment:
```suggestion
    if (taskIndex < 0 && dataJob !== this.constantsService.DELETE_TODO) {
      return
    }
    
    this.taskService.taskList.splice(taskIndex, 1);
```

Reply:
I would suggest to use the approach I provided. You can always modify/refactor it when needed. 

The reason we don't keep things like that is its easier to leave them there and lost track. 

Also early return is a good practice than leving them if and else block.

Reply:
done!